c10/core/TensorImpl.h

```diff
@@ -748,7 +764,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
 #ifdef DEBUG
```

Comment:
It seems like the invariant here is that (1) `numel_` is valid and (2) it's not symbolic (since `compute_numel` doesn't return symint). Therefore, should be unchecked. But it would be good to write down these invariants explicitly because it gets a little dicey with subclassing.